lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -4557,6 +4557,25 @@ FIRRTLType TailPrimOp::inferReturnType(ValueRange operands,
   return IntType::get(input.getContext(), false, width, inputi.isConst());
 }
 
+//===----------------------------------------------------------------------===//
+// Properties
+//===----------------------------------------------------------------------===//
+
+LogicalResult PathOp::verifyInnerRefs(hw::InnerRefNamespace &ns) {
+  auto targetRef = getTarget();
+  if (!targetRef)
+    return emitOpError("has invalid target reference");
+  if (targetRef.getModule() !=
+      (*this)->getParentOfType<FModuleLike>().getModuleNameAttr())
+    return emitOpError() << "has non-local target";
+
+  auto target = ns.lookup(targetRef);
```

Comment:
Want to suggest this checks the target's type (probably cause for utility to help) is indeed "hardware" but not sure how that'd really work since it could target operations themselves that do not have types (e.g., an instance).

Reply:
Yeah, I had this same thought, but decided if something implements InnerSymOpInterface, its probably safe to say its hardware - at least for now.